tests/Makefile.am

```diff
@@ -0,0 +1,71 @@
+# Copyright 2013-present Barefoot Networks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Unit tests
+TESTSUITE = $(srcdir)/tests/testsuite
+
+TESTSUITE_AT += \
+	tests/testsuite.at \
+	tests/frontend.at
+
+check-local: atconfig $(TESTSUITE) $(TESTSUITE_AT)
+	set $(SHELL) '$(TESTSUITE)' -C tests $(TESTSUITEFLAGS); \
+	"$$@" || (test X'$(RECHECK)' = Xyes && "$$@" --recheck)
+
+check-bmv2:
+	make check-local TESTSUITEFLAGS='-k bmv2 $(filter-out $@,$(MAKECMDGOALS))'
```

Comment:
This turns out to be a known issue in make 4.1, where the value N in -jN is not passed down to MAKEFLAGS. I installed make 4.2.1 from source, and problem solved.